bin/ci-check

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o noglob
+
+# Run all checks that CI needs to pass a build
+# Usage: bin/ci-check
+
+main() {
+  bin/format -al
```

Comment:
maybe add a `tsc` in here like @scotttrinh suggested?

Reply:
`bin/build` is executed later down the line but I can switch it out if you'd like.